examples/filters/colorfilter/color_filter.go

```diff
@@ -0,0 +1,162 @@
+// Package colorfilter implements a modular camera that filters the output of an underlying camera and only keeps
+// captured data if the vision service detects a certain color in the captured image.
+package colorfilter
+
+import (
+	"context"
+	"fmt"
+	"image"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"github.com/viamrobotics/gostream"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/services/vision"
+)
+
+var (
+	// Model is the full model definition.
+	Model            = resource.NewModel("example", "camera", "colorfilter")
+	errUnimplemented = errors.New("unimplemented")
+)
+
+func init() {
+	resource.RegisterComponent(camera.API, Model, resource.Registration[camera.Camera, *Config]{
+		Constructor: newCamera,
+	})
+}
+
+func newCamera(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (camera.Camera, error) {
+	c := &colorFilterCam{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+	if err := c.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// Config contains the name to the underlying camera and the name of the vision service to be used.
+type Config struct {
+	ActualCam     string `json:"actual_cam"`
+	VisionService string `json:"vision_service"`
+}
+
+// Validate validates the config and returns implicit dependencies.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.ActualCam == "" {
```

Comment:
Q: why is VisionService not required/an implicit dependency?

Reply:
Good catch - added!